components/tidb_query/src/batch/runner.rs

```diff
@@ -329,7 +332,12 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
         let mut warnings = self.config.new_eval_warnings();
         let mut ctx = EvalContext::new(self.config.clone());
 
+        let mut time_slice_start = Instant::now();
```

Comment:
How about using `now_coarse()`?

Reply:
I'm afraid Linux's `CLOCK_MONOTONIC_COARSE` cannot satisfy our requirements well.
Ticks on Linux are typically 100Hz. It means measuring a duration that is several milliseconds long can also fail.

Reply:
I write a small test: https://gist.github.com/sticnarf/5c25052ff8b8fba45203cda4c14f542f
Using `5000000` as the argument, a single loop should cost about 5ms. But if we measure it with the coarse way, the result is often zero.

Reply:
I can't reproduce your tests. I do see it zero while monotonic is several hundred us.

I also change your test into an infinite loop and output a log only when monotonic is not less than 1ms while coarse is less than 1ms. The test has been running silently for several minutes until I kill it.

Reply:
I also tried your second test yesterday and got the similar result as yours. I can just guess that some syscalls may also refresh the coarse one.

Reply:
According to https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux_for_real_time/7/html/reference_guide/sect-posix_clocks the coarse variant's resolution is 1ms and it should be enough.

Reply:
The resolution on Redhat Linux is about 1ms. It's different on other linux. For example, Ubuntu is about 4ms and the archlinux on my laptop is 3.3ms.

Reply:
A potential optimization is to obtain running time from thread pool.